### PR TITLE
LibWasm: Implement a bunch more WASI calls

### PR DESCRIPTION
**Pull request by alimpfard**
With this, my test case (aka zig.wasm) reads std.zig and then breaks somewhere trying to read from garbage memory :P

### CODE REVIEW

File: Userland/Libraries/LibWasm/WASI/Wasi.cpp [hunk 1]
```diff
@@ -588,7 +588,8 @@ ErrorOr<Result<void>> Implementation::impl$path_create_directory(Configuration&
     auto slice = TRY(slice_typed_memory(configuration, path, path_len));
     auto null_terminated_string = DeprecatedString::copy(slice);
 
-    if (mkdirat(dir_fd, null_terminated_string.characters(), 0755) < 0)
+    auto rc = mkdirat(dir_fd, null_terminated_string.characters(), 0755);
+    if (rc < 0)
```
Comment (GMTA) [inline Userland/Libraries/LibWasm/WASI/Wasi.cpp:602]: Not sure how this change is related; it also does not reflect the way you deal with error codes in a later commit (e.g. `if (fstat(...) < 0)`)

File: Userland/Libraries/LibWasm/WASI/Wasi.cpp [hunk 2]
```diff
@@ -1028,3 +1027,87 @@ Errno errno_value_from_errno(int value)
...
+                return Formatter<FormatString>::format(builder, "({}) {}"sv, Wasm::ValueType::kind_name(value.type().kind()), v);
+            });
+    }
+};
+
+template<>
+struct Formatter<Wasm::Wasi::Errno> : AK::Formatter<FormatString> {
+    ErrorOr<void> format(FormatBuilder& builder, Wasm::Wasi::Errno const& value)
+    {
+        return Formatter<FormatString>::format(builder, "{}"sv, (u16)value);
```
Comment (GMTA) [inline Userland/Libraries/LibWasm/WASI/Wasi.cpp:1137]: `static_cast<u16>(value)` pls

File: Userland/Libraries/LibWasm/WASI/Wasi.cpp [hunk 3]
```diff
@@ -697,6 +697,62 @@ ErrorOr<Result<Timestamp>> Implementation::impl$clock_time_get(Configuration&, C
...
+        if (S_ISBLK(buf.st_mode))
+            return FileType::BlockDevice;
+        if (S_ISREG(buf.st_mode))
+            return FileType::RegularFile;
+        if (S_ISFIFO(buf.st_mode))
+            return FileType::Unknown; // no Pipe? :yakfused:
+        if (S_ISLNK(buf.st_mode))
+            return FileType::SymbolicLink;
+        if (S_ISSOCK(buf.st_mode))
+            return FileType::SocketDGram; // :shrug:
```
Comment (GMTA) [inline Userland/Libraries/LibWasm/WASI/Wasi.cpp:740]: Either these two comments for FIFO and sockets do not explain a lot (-> delete) or they should be `FIXME`s

Comment (alimpfard) [reply Userland/Libraries/LibWasm/WASI/Wasi.cpp:740]: It's not something we can fix (WASI spec missing "proper" enum values), and it deserves a comment imo.
I should probably put _that_ in the comment :sweat_smile:

**Comment by alimpfard** [comment]
ohhhh I was gonna look at this, right right. thank you stalebot.

**Pull request merged**